### PR TITLE
PR: Fix conda env create handling for remote specs

### LINKED ISSUE #7755: fix 'conda env create` using remote env file
Fix `conda env create  mfix/mfix-18.1.0-osx64` in master. Broken in https://github.com/conda/conda/pull/7385.

xref: https://github.com/conda/conda/issues/7111

Can also remove the deprecation warnings now (they were apparently placed in the wrong code location to begin with).

### PR DESCRIPTION
**Pull request by goanpeca**
Fixes #7755

### CODE REVIEW

**Comment by goanpeca** [comment]
@kalefranz some of the tests fails are not related to this PR however when trying this locally

I have some issues with https://github.com/conda/conda/blob/master/tests/conda_env/test_cli.py#L205 but it seems to point to an issue on the conda-env (deprecated) repo which I cannot access. Thoguths?

**Comment by goanpeca** [comment]
@kalefranz I have added a test.

**Comment by kalefranz** [comment]
The issue 254 reference was

> Previously we could override the name from an environment file using `--name` in the CLI, but this is no longer respected after the latest release. This was very useful for heroku buildpacks.

**Comment by kalefranz** [comment]
I think there might be one appveyor test failure in master right now, but I think master is fine for linux tests.  Maybe you need to pull and rebase?

File: tests/conda_env/test_cli.py [hunk 1]
```diff
@@ -203,12 +218,24 @@ def test_update(self):
...
         except Exception as e:
-            self.assertIsInstance(e, EnvironmentFileNotFound, str(e))
+            print(e)
+
+        o, e = run_conda_command(Commands.INFO, None, "--json")
+
+        parsed = json.loads(o)
+        self.assertNotEqual(
+            len([env for env in parsed['envs'] if env.endswith(env_name)]), 0
+        )
```
Comment (goanpeca) [inline tests/conda_env/test_cli.py]: I am not sure what this test was actually doing, but I updated it to check @kalefranz mentioned the issue was.

File: tests/conda_env/test_cli.py [hunk 2]
```diff
@@ -86,16 +87,16 @@ def run_env_command(command, prefix, *arguments):
...
-            command_line = "{0} ".format(command)
+            command_line = "{0} {1}".format(command, " ".join(arguments))
     elif command is Commands.ENV_REMOVE:  # REMOVE
         command_line = "{0} --yes -n {1} {2}".format(command, prefix, " ".join(arguments))
     elif command is Commands.ENV_UPDATE:
         command_line = "{0} -n {1} {2}".format(command, prefix, " ".join(arguments))
     else:
         command_line = " --help "
-
+    print(command_line)
```
Comment (jjhelmus) [inline tests/conda_env/test_cli.py]: probably is not necessary but if fine inside a test

**Comment by jjhelmus** [comment]
LGTM, failing tests are unrelated to changes.

**Pull request merged**